Toastify/src/Core/ApplicationStartupException.cs

```diff
@@ -5,44 +5,42 @@ namespace Toastify.Core
 {
     public class ApplicationStartupException : ApplicationException
     {
-        public ApplicationStartupException() : this(true)
+        public ApplicationStartupException(string message) : base(message + Environment.NewLine + CreateMessage())
         {
         }
 
-        public ApplicationStartupException(bool spotifyStatus) : base(CreateMessage(spotifyStatus))
+        private static string CreateMessage()
         {
-        }
-
-        public ApplicationStartupException(string message) : this(message, true)
-        {
-        }
+            var nl = Environment.NewLine;
 
-        public ApplicationStartupException(string message, bool spotifyStatus) : base($"{CreateMessage(spotifyStatus)}\nAdditional details:\n{message}")
-        {
-        }
+            // Touching Spotify.Instance forces instance creation, but it fails if called too early
+            try
+            {
+                var tmp = Spotify.Instance;
+            }
+            catch (NullReferenceException)
+            {
+                return "Spotify instance status: not yet created.";
+            }
 
-        private static string CreateMessage(bool spotifyStatus = true)
-        {
             List<string> messages = new List<string>();
+            messages.Add("Spotify instance running status: " + Spotify.Instance.IsRunning);
 
-            if (spotifyStatus)
+            var status = Spotify.Instance.Status;
+            messages.Add("Spotify instance Status: ");
+            if (status == null)
+                messages.Add("null");
+            else
             {
-                bool spotifyInstanceCreated = Spotify.Instance != null;
-                if (spotifyInstanceCreated)
-                {
-                    bool spotifyIsRunning = Spotify.Instance.IsRunning;
-                    messages.Add($"Spotify is running: {spotifyIsRunning}");
-
-                    var status = Spotify.Instance.Status;
-                    messages.Add(status != null
-                        ? $"Status: {status.Online}, {status.Running}, {status.Version}, {status.ClientVersion}, {status.Track != null}"
-                        : "Status: null");
-                }
-                else
-                    messages.Add("Spotify instance not created!");
+                // TODO: add a status.toString() to Spotify API and replace this:
+                messages.Add("-- Online: " + status.Online);
+                messages.Add("-- Running: " + status.Running);
+                messages.Add("-- Version: " + status.Version);
+                messages.Add("-- Client Version: " + status.ClientVersion);
+                messages.Add("--  has Track: " + (status.Track != null));
```

Comment:
Do you mind using string interpolation instead of concatenation? I find it much more readable.

Reply:
Was using it first, but then decided against it when I saw that it is basically a single concatenation and not really worth using template strings. But I'll revert it.